docs/Contributing/API-for-contributors.md

```diff
@@ -536,54 +536,113 @@ The MDM endpoints exist to support the related command-line interface sub-comman
-}
-```
+`Status: 204`
 
-Note that the `public_key` and `private_key` are base64 encoded and should be decoded before writing them to files.
 
 ### Request Certificate Signing Request (CSR)
 
-`POST /api/v1/fleet/mdm/apple/request_csr`
+`GET /api/v1/fleet/mdm/apple/request_csr`
 
 #### Parameters
 
 | Name          | Type   | In   | Description                                                                            |
 | ------------- | ------ | ---- | -------------------------------------------------------------------------------------- |
-| email_address | string | body | **Required.** The email that will be associated with the Apple APNs certificate.       |
-| organization  | string | body | **Required.** The name of the organization associated with the Apple APNs certificate. |
+| alt  | string | query | **Required.** Set to "media" to download the certificate signing request (CSR). |
 
-#### Example
+#### Example (download CSR)
 
-`POST /api/v1/fleet/mdm/apple/request_csr`
+`GET /api/v1/fleet/mdm/apple/request_csr?alt=media`
 
 ##### Default response
 
-```json
-{
-  "apns_key": "aGV5LCBJJ20gc2VjcmV0Cg==",
-  "scep_cert": "bHR5LCBJJ20gc2VjcmV0Cg=",
-  "scep_key": "lKT5LCBJJ20gc2VjcmV0Cg="
-}
+```
+Status: 204
+```
```

Comment:
@jahzielv @marko-lisica heads-up I think we want this to be a `200` with:

- the CSR contents as the body
- a `Content-Disposition` header

we could also drop the `alt=media` param and return the CSR base64 encoded in the JSON response (this is how the current endpoint works)

```json
{
    "csr": "lKT5LCBJJ20gc2VjcmV0Cg="
}

Reply:
@roperzh Would we be able to download CSR in the UI? Can we take this response on the frontend and create file that can be downloaded?

Reply:
@marko-lisica yes, that's currently how the UI works! no strong preference for any as long as we adjust this to return the CSR in some form 👍

Reply:
If that's what we do in UI today, I guess it'll be easier to implement this if we return JSON.

Reply:
agree! sounds good to me too

Reply:
I'll update PR

Reply:
@marko-lisica thanks! in that case we should update the ABM endpoint with the same!